lib/reporter/reporters/html.js

```diff
@@ -76,50 +38,204 @@ class HtmlReporter extends BaseReporter {
+      const modules = modulesWithEnv[testEnv];
 
-    let rendered = ejs.render(data, results, {
-      views: [sourceHtmlFolder]
-    });
-    rendered = Utils.stripControlChars(rendered);
+      for (const moduleKey of Object.keys(modules)) {
+        const module = modules[moduleKey];
+  
+  
+        const {passedCount, failedCount} = HtmlReporter.computePassedAndFailedCounts(module);
+        
+        if (!environments[testEnv]) {
+          environments[testEnv] = {
+            metadata: {
+              'platformName': module.sessionCapabilities.platformName,
+              'device': 'desktop',
+              'browserName': module.sessionCapabilities.browserName,
+              'browserVersion': module.sessionCapabilities.browserVersion,
+              'executionMode': 'local'
+            },
+            stats: {
+              'passed': passedCount,
+              'failed': failedCount,
+              'skipped': module.skippedCount,
+              'total': passedCount + failedCount + module.skippedCount,
+              'time': module.timeMs 
+            },
+            modules: {}
+          };
+        } else {
+          environments[testEnv].stats.passed += passedCount;
+          environments[testEnv].stats.failed += failedCount;
+          environments[testEnv].stats.skipped += module.skippedCount;
+          environments[testEnv].stats.total += passedCount + failedCount + module.skippedCount;
+          environments[testEnv].stats.time = environments[testEnv].stats.time || module.timeMs;
+          environments[testEnv].metadata.platformName = environments[testEnv].metadata.platformName || module.sessionCapabilities.platformName;
+          environments[testEnv].metadata.browserName = environments[testEnv].metadata.browserName || module.sessionCapabilities.browserName;
+          environments[testEnv].metadata.browserVersion = environments[testEnv].metadata.browserVersion || module.sessionCapabilities.browserVersion;
+        }
+ 
+        environments[testEnv].modules[moduleKey] = HtmlReporter.adaptModule(module);
+      }
+    }
+
+    return environments;
+  }
+
+  static aggregateStats(environments) {
+    const stats = {
+      total: 0,
+      passed: 0,
+      failed: 0,
+      skipped: 0,
+      time: 0
+    };
+    
+    for (const envName of Object.keys(environments)) {
+      const env = environments[envName];
+
+      stats.passed += env.stats.passed;
+      stats.failed += env.stats.failed;
+      stats.skipped += env.stats.skipped;
+      stats.total += env.stats.total;
+      stats.time += env.stats.time || 0;
+    }
+
+    return stats;
+  }
+
+  static adaptModule(module) {
 
-    return this.writeReportFile(filename, rendered, destFolder, sourceAssetsFolder)
-      .then(_ => {
-        Logger.logDetailedMessage(Logger.colors.stack_trace(` Wrote HTML report file to: ${path.resolve(filename)}` + '\n'), 'info');
+    // Pick only the necessary fields
+    const result =  {
```

Comment:
couldn't this be written in a more elegant manner?

Reply:
Is using `lodash.pick` a better option? Or is there some other way I can use?